warehouse/routes.py

```diff
@@ -296,6 +296,9 @@ def includeme(config):
     )
     config.add_route("packaging.file", files_url)
 
+    # Re-Authentication Route
+    config.add_route("reauthenticate", "/reauthenticate/", domain=warehouse)
+
```

Comment:
I'm thinking this should probably be mounted under `/account/reauthenticate` instead, which will ensure that we won't have any cache issues because everything under `/account` is not cached.

This means that this should move up where the rest of the `/account` routes are defined above, and the view should move into `warehouse/accounts/views.py`, and tests as appropriate.